components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/username/NotificationUsernameRecoveryManager.java

```diff
@@ -209,3 +101,138 @@
 //        }
 //    }
 //}
+
+    public boolean recoverUserByClaims(ArrayList<UserClaim> claims)
+            throws IdentityRecoveryException {
+
+        // Check whether username recovery enable
+        boolean isRecoveryUsernameEnable = Utils.getRecoveryConfigs().getRecovery().getUsername().isEnable();
```

Comment:
Default it is enabled in the config file. If a client wants that he can disable it. DoI need to check null?